viewing_party/party.py

```diff
@@ -0,0 +1,150 @@
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    all_ratings = 0
+    if len(user_data["watched"]) == 0: 
+        print("We haven't started watching yet!")
+        return 0
+    for movie in user_data["watched"]:
+        all_ratings += movie["rating"]
+    return all_ratings / len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    if len(user_data["watched"]) == 0: 
+        print("We haven't started watching yet!")
+        return None
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_dict:
+            genre_dict[movie["genre"]] = 1
+        else:
+            genre_dict[movie["genre"]] += 1
+
+    genre_tally = 0
+    fave_genre = ""
+    for genre, count in genre_dict.items():
+        if count > genre_tally:
+            genre_tally = count
+            fave_genre = genre
+    return fave_genre
+
+
+#wave 3
+def get_unique_watched(user_data):
+    only_friends_watched =[]
+    my_unique_watched = []
+    for watched in user_data["friends"]:
+        for movie in watched["watched"]:
+            only_friends_watched.append(movie["title"])
+    for movie in user_data["watched"]:
+        if movie["title"] not in only_friends_watched:
+            my_unique_watched.append(movie)
+    return my_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    their_movies = []
+    for movie in user_data["watched"]:
+        their_movies.append(movie)
+
+    friends_watched = []
+    for friends in user_data["friends"]:
+        for movie in friends["watched"]:
+            if movie not in user_data["watched"]:
+                friends_watched.append(movie)
+
+    friends_unique_watched = []
+    for movie in friends_watched:
+        if movie not in user_data["watched"]:
+            if movie not in friends_unique_watched:
+                friends_unique_watched.append(movie)
+    return friends_unique_watched
+
+
+def approved_user_subs(host, subscriptions):
+    if host in subscriptions:
+        return True
+    else:
+        return False
+
+
+def get_friends_movies(user_data):
+    friends_titles = set()
+    friends = user_data["friends"]
+    friends_movies = []
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie["title"] not in friends_titles:
+                friends_movies.append(movie)
+                friends_titles.add(movie["title"])
+    return friends_movies
+
+#wave 4
+def get_available_recs(user_data):
+    friends_recs = []
+    my_rec_list = []
+    user_watched = []
+    user_subs = user_data["subscriptions"]
+    for friend in user_data["friends"]:
+        for film in friend["watched"]:
+            friends_recs.append(film)
+    for item in user_data["watched"]:
+        user_watched.append(item["title"])
+        user_subs = user_data["subscriptions"]
```

Comment:
This might be repeat code. You assigned that subscriptions to `user_subs` on line 117. I don't think you need to re-assign the list of subscriptions every iteration.

```suggestion

```